### ISSUE TITLE
future cannot be sent between threads safely

### ISSUE THREAD

**Issue by cobbinma** [awslabs/aws-sdk-rust#224]
### Version

```toml
aws-sdk-s3 = { git = "ssh://github.com/awslabs/aws-sdk-rust.git", tag = "v0.0.17-alpha", package = "aws-sdk-s3" }
```

### Platform

`Darwin 0349 20.6.0 Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:27 PDT 2021; root:xnu-7195.141.2~5/RELEASE_ARM64_T8101 arm64`

### AWS Services

`aws-sdk-s3`

### Description

hi all 👋 

I want to use aws s3 in an asynchronous application however i'm getting this error when trying to use it in an `async_trait`

The trait `Send` is not implemented for `(dyn smithy_eventstream::frame::UnmarshallMessage<Error = SelectObjectContentError, Output = SelectObjectContentEventStream> + 'static)`

Is this an issue with the sdk types? Is there a workaround for the code below?

I tried this code:

```rust
use aws_sdk_s3::model::ExpressionType;
use aws_sdk_s3::Client;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    Ok(())
}

struct MyClient {
    client: Client
}

#[async_trait::async_trait]
impl Test for MyClient {
    async fn test(&self) -> Result<(), Box<dyn std::error::Error>> {
        let mut output = self.client
        .select_object_content()
        .bucket("bucket")
        .key("object.csv")
        .expression_type(ExpressionType::Sql)
        .expression("SELECT * from S3Object")
        .send()
        .await?;

    while let Some(event) = output.payload.recv().await? {
        match event {
            _ => (),
        }
    }

    Ok(())
    }
}

#[async_trait::async_trait]
trait Test {
    async fn test(&self) -> Result<(), Box<dyn std::error::Error>>;
}
```

Error:

```
cargo run                 
   Compiling s3-select-rs v0.1.0 (/Users/matthew.cobbing/go/src/git.perkbox.io/matthew.cobbing/s3-select-rs)
error: future cannot be sent between threads safely
  --> src/main.rs:15:68
   |
15 |       async fn test(&self) -> Result<(), Box<dyn std::error::Error>> {
   |  ____________________________________________________________________^
16 | |         let mut output = self.client
17 | |         .select_object_content()
18 | |         .bucket("bucket")
...  |
31 | |     Ok(())
32 | |     }
   | |_____^ future created by async block is not `Send`
   |
   = help: the trait `Send` is not implemented for `(dyn smithy_eventstream::frame::UnmarshallMessage<Error = SelectObjectContentError, Output = SelectObjectContentEventStream> + 'static)`
note: future is not `Send` as this value is used across an await
  --> src/main.rs:25:29
   |
25 |     while let Some(event) = output.payload.recv().await? {
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ first, await occurs here, with `output.payload` maybe used later...
note: `output.payload` is later dropped here
  --> src/main.rs:29:5
   |
25 |     while let Some(event) = output.payload.recv().await? {
   |                             -------------- has type `smithy_http::event_stream::Receiver<SelectObjectContentEventStream, SelectObjectContentError>` which is not `Send`
...
29 |     }
   |     ^
help: consider moving this into a `let` binding to create a shorter lived borrow
  --> src/main.rs:25:29
   |
25 |     while let Some(event) = output.payload.recv().await? {
   |                             ^^^^^^^^^^^^^^^^^^^^^
   = note: required for the cast to the object type `dyn Future<Output = Result<(), Box<(dyn std::error::Error + 'static)>>> + Send`

error: aborting due to previous error

error: could not compile `s3-select-rs`

To learn more, run the command again with --verbose.
```

**Comment by rcoh** [comment]
thanks for the report! we'll get this fixed.

**Comment by cobbinma** [comment]
Thanks @rcoh, don't think that will work for us as we're implementing generated traits. Will wait for the fix 😄

**Comment by rcoh** [comment]
reopening pending the release of v0.0.19

**Comment by jdisanti** [comment]
The fix for this landed in [`v0.0.19-alpha`](https://github.com/awslabs/aws-sdk-rust/releases/tag/v0.0.19-alpha). Thank you!
